File: 240513/괄호 쌍 만들어주기 2/pair-parentheses-2.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        String str = sc.nextLine();
        char[] arr = str.toCharArray();

        int len = str.length();
        int count = 0;
        for(int i=0;i<len-3;i++){
            if(arr[i] == '(' && arr[i+1] == '('){
                for(int j=i+2;j<len-1;j++) {
                    if(arr[j]==')' && arr[j+1]==')')
                        count++;
                }
            }
        }
        System.out.println(count);
    }
}

File: 240504/괄호 쌍 만들어주기 2/pair-parentheses-2.kt
import java.util.Scanner

fun main() {
    val sc = Scanner(System.`in`)

    val input = sc.next()
    var result = 0

    for (i in 0 until input.length) {
        for (j in i + 1 until input.length) {
            for (k in j + 1 until input.length) {
                for (l in k + 1 until input.length) {
                    if (input[i] == '(' && input[j] == '(' && i + 1 == j && k + 1 == l && input[k] == ')' && input[l] == ')') {
                        result++
                    }
                }
            }
        }
    }
    
    print(result)
}